src/hotspot/cpu/aarch64/c1_LIRAssembler_aarch64.cpp

```diff
@@ -2558,6 +2558,11 @@ void LIR_Assembler::emit_lock(LIR_OpLock* op) {
   Register hdr = op->hdr_opr()->as_register();
   Register lock = op->lock_opr()->as_register();
   if (UseHeavyMonitors) {
+    if (op->info() != NULL) {
+      int null_check_offset = __ offset();
+      __ null_check(obj, -1);
```

Comment:
```suggestion
      __ null_check(obj);
```

Reply:
Yes, that would be better but it leads to a compilation error on macosx-aarch64 that null_check is ambiguous:

src/hotspot/cpu/aarch64/c1_LIRAssembler_aarch64.cpp:2563:10: error: call to member function 'null_check' is ambiguous
[2022-06-30T00:09:19,094Z]       __ null_check(obj);
[2022-06-30T00:09:19,094Z]       ~~~^~~~~~~~~~
src/hotspot/cpu/aarch64/macroAssembler_aarch64.hpp:602:16: note: candidate function
[2022-06-30T00:09:19,094Z]   virtual void null_check(Register reg, int offset = -1);
[2022-06-30T00:09:19,094Z]                ^
src/hotspot/cpu/aarch64/c1_MacroAssembler_aarch64.hpp:109:8: note: candidate function
[2022-06-30T00:09:19,094Z]   void null_check(Register r, Label *Lnull = NULL) { MacroAssembler::null_check(r); }
[2022-06-30T00:09:19,094Z]        ^

Reply:
OK, nevermind.